app/controllers/immunizations_controller.rb

```diff
@@ -0,0 +1,20 @@
+class ImmunizationsController < ApplicationController
+  def update
+    if admin?
```

Comment:
I would consider a callback for all acions that require admin rights. It will make the controller much cleaner and we will avoid ifs-hell 😈